File: Adjust/examples/example-app-java/src/main/java/com/adjust/examples/ServiceActivity.java
package com.adjust.examples;

import android.content.Intent;
import android.net.Uri;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import com.adjust.sdk.Adjust;
import com.adjust.sdk.AdjustDeeplink;

public class ServiceActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_service);

        Intent intent = getIntent();
        Uri data = intent.getData();
        AdjustDeeplink adjustDeeplink = new AdjustDeeplink(data);
        Adjust.processDeeplink(adjustDeeplink, getApplicationContext());
    }

    public void onServiceClick(View v) {
        Intent intent = new Intent(this, ServiceExample.class);
        startService(intent);
    }

    public void onReturnClick(View v) {
        finish();
    }
}

File: Adjust/examples/example-app-kotlin/src/main/java/com/adjust/examples/ServiceActivity.kt
package com.adjust.examples

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View

import com.adjust.sdk.Adjust
import com.adjust.sdk.AdjustDeeplink

class ServiceActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_service)

        val intent = intent
        val data = intent.data
        val adjustDeeplink = AdjustDeeplink(data);
        Adjust.processDeeplink(adjustDeeplink, applicationContext)
    }

    fun onServiceClick(v: View) {
        val intent = Intent(this, ServiceExample::class.java)
        startService(intent)
    }

    fun onReturnClick(v: View) {
        finish()
    }
}